### DISCUSSION TITLE
Trouble with selectors

### DISCUSSION THREAD

**Discussion by geiseri** [CadQuery/cadquery#1155] [Q&A]
Hi, I am struggling with trying to select a face on an object so I can create a workplane.  In my case I have a box as below: 
![image](https://user-images.githubusercontent.com/2265693/190410739-ce4bec0e-caa5-4e70-82bb-73a24ad898b2.png)

I get the face I want with the selector `case_bottom.faces("<X[1]")`, but what I really want is to get the "<X" axis face with one edge on ">Z". 

Now I might be going about this wrong.  I am putting a "snap fit" on that face, so later on my selector no longer works since the "<X" indices have changed.  Any advice here?  Thanks!

**Comment by jmwright** [comment]
@geiseri I'm not sure what you mean by a "snap fit" in this context. Are you adding a ridge or detent or something like that?

If you can share your full code, or at least more of it, it might be easier to help.

**Reply by geiseri** [reply]
By "snapfit" I have two surfaces that overlap, one side with a groove and the other with bump that is small enough to slide over and when the two parts line up the bump secures it in the groove. In oscad I created a cylinder and cut it out of one side and unioned on the other one. Does that make sense?
